xml_converter/src/xml_converter.cpp

```diff
@@ -33,46 +34,58 @@
 
 using namespace std;
 
-map<string, vector<string>> read_taco_directory(
+struct CategoryWithinSinglePack {
+    Category* category;
+    vector<string> file_paths;
+};
+
+struct CategoryWithinMultiplePacks {
+    Category* category;
+    vector<vector<string>> file_path_sets;
+};
+
+map<Attribute::UniqueId::UniqueId, CategoryWithinSinglePack> read_taco_directory(
     string input_path,
     map<string, Category>* marker_categories,
     vector<Parseable*>* parsed_pois
 ) {
-    map<string, vector<string>> top_level_category_file_locations;
+    map<Attribute::UniqueId::UniqueId, CategoryWithinSinglePack> top_level_category_file_locations;
     if (!filesystem::exists(input_path)) {
         cout << "Error: " << input_path << " is not an existing directory or file" << endl;
     }
 
     string directory_name = filesystem::path(input_path).filename();
     vector<MarkerPackFile> xml_files = get_files_by_suffix(input_path, ".xml");
     for (const MarkerPackFile& path : xml_files) {
-        set<string> top_level_category_names = parse_xml_file(path, marker_categories, parsed_pois);
+        map<Attribute::UniqueId::UniqueId, Category*> top_level_categories = parse_xml_file(path, marker_categories, parsed_pois);
         string file_path = join_file_paths(input_path, path.relative_filepath);
-        for (set<string>::iterator it = top_level_category_names.begin(); it != top_level_category_names.end(); it++) {
-            top_level_category_file_locations[*it].push_back(file_path);
+        for (map<Attribute::UniqueId::UniqueId, Category*>::iterator it = top_level_categories.begin(); it != top_level_categories.end(); it++) {
+            top_level_category_file_locations[it->first].category = it->second;
```

Comment:
The existing an new pointers should be the same, should we be sanity checking that here?
It will require a little bit of extra logic because the first iteration will have garbage in `.category` so we can't compare on every iteration.
I worry mostly about a possible edge case where the pointer is not the same for some reason and it gets overwritten causing unknown behavior.